custom_metrics/lazy-in-viewport.js

```diff
@@ -0,0 +1,11 @@
+// [lazy-in-viewport]
+// Uncomment the previous line for testing on webpagetest.org
+
+let images = document.querySelectorAll("img[loading=lazy]");
+let lazyImages = [];
+images.forEach((img) => {
+  if (img.getBoundingClientRect().top < window.innerHeight) {
+    lazyImages.push(img.src);
```

Comment:
How do you plan on using the `src` in your analysis? I wonder if it'd be useful to have other metadata about the image.

Also not sure whether this metric should live in its own JS file or combined with another related one, either `almanac.js` or we could start `resource_hints.js` and keep adding to it. WDYT?

Reply:
I don't need the `src` actually and would be happy with a count (`array.length`). This way I would be able to say

> 10% of all lazy loaded images are actually within the initial viewport.

Reply:
> Also not sure whether this metric should live in its own JS file or combined with another related one, either almanac.js or we could start resource_hints.js and keep adding to it. WDYT?

I can add it to the `images` custom metric already being used in `almanac.js`. That's your call. I was initially told that separate files are preferred, but either works for me.

Reply:
I think WPT includes a custom metric by default which includes the `src` of each image and seems to work well with `picture` too.

```
[
  {
    "url": "https://googlechrome.github.io/samples/picture-element/images/kitten-large.png",
    "width": 650,
    "height": 285,
    "naturalWidth": 650,
    "naturalHeight": 285
  },
  {
    "url": "https://googlechrome.github.io/samples/picture-element/images/butterfly.webp",
    "width": 320,
    "height": 320,
    "naturalWidth": 320,
    "naturalHeight": 320
  }
]
```

I think we should be able to use it to answer questions on the file type, domain & whether it has been preloaded. Unless you want to segment this data for lazy-loaded images.

Reply:
There are no default custom metrics. This is 
 this one: https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/Images.js

I think you should you add `loading` attributes to that as well as an `in-viewport` boolean, rather than defining a new one?

Reply:
Actually I see we copied that Images.js from WPT and that still runs on the public instances.

@rviscomi / @pmeenan any thoughts on whether we should leave that alone then so we're still in sync, and define Almanac specific changes to a separate file (either the existing almanac.js or a new resource-hints.js)? Might be confusing if we diverge image.js...

Reply:
I'd be ok modifying the images metric, parity with WPT isn't a requirement. @pmeenan any other thoughts?

Reply:
The WPT metric is just for the Cloudinary image analysis integration which doesn't really apply to the HTTP Archive so feel free to modify it or completely replace it.

Reply:
Based on the above I have modified the Images custom metric and updated the first comment with a link to the updated test and data.

Note that there is a bit of overlap between the Images custom metric and the `images` property in the `almanac.js`. I have not updated the latter.